### PR TITLE
[cli] Added fetch dist-tags http status validation

### PR DESCRIPTION
**Pull request by cb1kenobi**
If for some reason the `get-latest-worker.js` fails to get the dist-tags, it doesn't check the status code and proceeds to `JSON.parse()` the response. If for example npm returns 401 Unauthorized, you get a cryptic JSON parse error message.

### CODE REVIEW

File: packages/cli/src/util/get-latest-version/get-latest-worker.js [hunk 1]
```diff
@@ -247,6 +247,14 @@ async function fetchDistTags(name) {
         });
         res.on('end', () => {
           try {
+            if (res.statusCode && res.statusCode >= 400) {
+              return reject(
```
Comment (Ethan-Arrowood) [inline packages/cli/src/util/get-latest-version/get-latest-worker.js:251]: Don't think you need the return here

**Review by Ethan-Arrowood — APPROVED** [review]

File: packages/cli/src/util/get-latest-version/get-latest-worker.js (hunk 1 above)
Comment (cb1kenobi) [reply packages/cli/src/util/get-latest-version/get-latest-worker.js:251]: This `return` will prevent the rejected promise from attempting to resolve/reject later. To my knowledge, `reject()` does not stop the execution of the current scope.

File: packages/cli/src/util/get-latest-version/get-latest-worker.js [hunk 2]
```diff
@@ -247,6 +247,14 @@ async function fetchDistTags(name) {
         });
         res.on('end', () => {
           try {
+            if (res.statusCode && res.statusCode >= 400) {
```
Comment (styfle) [inline packages/cli/src/util/get-latest-version/get-latest-worker.js:250]: Can you use (!res.ok) instead?

Comment (cb1kenobi) [reply packages/cli/src/util/get-latest-version/get-latest-worker.js:250]: This is using Node's built-in `https.get()` in which the handler receives a `ServerResponse` and that response object doesn't have an `ok` property.

**Review by styfle — APPROVED** [review]

File: packages/cli/src/util/get-latest-version/get-latest-worker.js (hunk 1 above)
Comment (Ethan-Arrowood) [reply packages/cli/src/util/get-latest-version/get-latest-worker.js:251]: You're correct, it doesn't stop the execution, but I don't believe you can resolve after a promise rejects

Comment (EndangeredMassa) [reply packages/cli/src/util/get-latest-version/get-latest-worker.js:251]: I'd rather not allow further execution that happens to not do anything. I think it's more correct and less brittle to return here.

**Pull request merged**